neuralhydrology/nh_run.py

```diff
@@ -67,8 +67,10 @@ def start_run(config_file: Path, gpu: int = None):
     config = Config(config_file)
 
     # check if a GPU has been specified as command line argument. If yes, overwrite config
-    if gpu is not None:
+    if gpu is not None and gpu >= 0:
         config.device = f"cuda:{gpu}"
+    if gpu is not None and gpu < 0:
```

Comment:
You explained me the reason on Skype, so I agree with that addition. However, would be good if you could also add the doc-string appropriately (here and in all other functions in this file where you changed it) as well as in the help string of the `_get_args()` function for the gpu argument.